hkm/views/views.py

```diff
@@ -475,27 +475,61 @@ def get_template_names(self):
     def setup(self, request, *args, **kwargs):
         # Session expires after browser is closed
         request.session.set_expiry(0)
-        self.search_term = request.GET.get('search', '')
-        self.page = int(request.GET.get('page', 1))
+        self.url_params = {
+            "search": request.GET.get('search', ''),
+            "page": int(request.GET.get('page', 1)),
+            "date_from": request.GET.get('date_from', ''),
+            "date_to": request.GET.get('date_to', ''),
+            "author": filter(None, request.GET.getlist('author', None)),
+            "date": filter(None, request.GET.getlist('date', None))
+        }
```

Comment:
Not really sure if constructing an object from the url params was the best option, but this makes the url param handling in tags much easier.